include/ginkgo/core/preconditioner/ic.hpp

```diff
@@ -402,22 +402,30 @@ class Ic : public EnableLinOp<Ic<LSolverTypeOrValueType, IndexType>>,
         // If no factories are provided, generate default ones
         if (!parameters_.l_solver_factory) {
             if constexpr (std::is_same_v<l_solver_type, LinOp>) {
```

Comment:
The if branches can be done inline:
```suggestion
            l_solver_ = generate_default_solver<std::conditional_t<
                std::is_same_v<l_solver_type, LinOp>,
                solver::LowerTrs<value_type, index_type>, l_solver_type>>(
                exec, l_factor);
```